### PR TITLE
Support to enable TLS for ng monitoring

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Support to enable TLS for ng monitoring
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 

Deploy tidb cluster that enable TLS, and deploy ng monitoring.

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by csuzhangxc — APPROVED** [review]
Will you add any E2E cases about TNGM TLS?

**Comment by KanShiori** [comment]
> Will you add any E2E cases about TNGM TLS?

yes, add E2E about ngm in the future 😄

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@KanShiori UT failed.

File: pkg/manager/tidbngmonitoring/ng_monitoring_manager_test.go [hunk 1]
```diff
@@ -258,33 +228,26 @@ func TestNewNGMonitorManager(t *testing.T) {
...
-			// mock result of status update
-			populateStatusPatch := gomonkey.ApplyPrivateMethod(reflect.TypeOf(manager), "populateStatus", func(_ *ngMonitoringManager, _ *v1alpha1.TidbNGMonitoring, _ *apps.StatefulSet) error {
-				return testcase.populateStatusErr
-			})
-			defer populateStatusPatch.Reset()
-			// mock result of configmap update
-			syncConfigMapPatch := gomonkey.ApplyPrivateMethod(reflect.TypeOf(manager), "syncConfigMap", func(_ *ngMonitoringManager, _ *v1alpha1.TidbNGMonitoring, _ *apps.StatefulSet) (*corev1.ConfigMap, error) {
-				return nil, testcase.syncConfigMapErr
-			})
-			defer syncConfigMapPatch.Reset()
```
Comment (KanShiori) [inline pkg/manager/tidbngmonitoring/ng_monitoring_manager_test.go:287]: The func `gomonkey.ApplyPrivateMethod` don't work for go1.17, so I removed two test case.

**Pull request merged**